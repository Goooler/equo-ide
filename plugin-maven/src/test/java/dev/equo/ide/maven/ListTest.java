/*******************************************************************************
 * Copyright (c) 2022 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.ide.maven;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import java.io.IOException;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
public class ListTest extends MavenHarness {
	@Test
	public void help(Expect expect) throws IOException, InterruptedException {
		setPom("");
		mvnw("help:describe -Dcmd=equo-ide:list -Ddetail")
				.snapshotBetween("Mojo: 'equo-ide:list'", "[INFO] BUILD SUCCESS", expect);
	}

	@Test
	public void argErrorCheck(Expect expect) throws IOException, InterruptedException {
		setPom("");
		mvnw("equo-ide:list")
				.snapshotBetween(
						" on project equo-maven-test-harness: ", "-> [Help 1]", expect.scenario("no args"));
		mvnw("equo-ide:list -Dinstalled -Dproblems")
				.snapshotBetween(
						" on project equo-maven-test-harness: ", "-> [Help 1]", expect.scenario("multi args"));
	}

	@Test
	public void defaultP2(Expect expect) throws IOException, InterruptedException {
		setPom("");
		mvnw("equo-ide:list -Dinstalled")
				.snapshotBetween(
						"(default-cli) @ equo-maven-test-harness ---",
						"[INFO] BUILD SUCCESS",
						expect.scenario("installed"));
		mvnw("equo-ide:list -Dproblems")
				.snapshotBetween(
						"(default-cli) @ equo-maven-test-harness ---",
						"[INFO] BUILD SUCCESS",
						expect.scenario("problems"));
		mvnw("equo-ide:list -Doptional")
				.snapshotBetween(
						"(default-cli) @ equo-maven-test-harness ---",
						"[INFO] BUILD SUCCESS",
						expect.scenario("optional"));
	}
}
