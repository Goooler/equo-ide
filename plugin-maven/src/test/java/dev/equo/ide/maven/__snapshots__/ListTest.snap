dev.equo.ide.maven.ListTest.argErrorCheck[multi args]=[
Exactly one of -Dinstalled, -Dproblems, -Doptional, -Dall=[categories|features|jars], -Ddetail=id, or -Draw=id must be set.
[ERROR] `mvn help:describe -Dcmd=equo-ide:list -Ddetail` for more info or visit https://github.com/equodev/equo-ide/blob/main/P2_MULTITOOL.md
[ERROR]
]


dev.equo.ide.maven.ListTest.argErrorCheck[no args]=[
Exactly one of -Dinstalled, -Dproblems, -Doptional, -Dall=[categories|features|jars], -Ddetail=id, or -Draw=id must be set.
[ERROR] `mvn help:describe -Dcmd=equo-ide:list -Ddetail` for more info or visit https://github.com/equodev/equo-ide/blob/main/P2_MULTITOOL.md
[ERROR]
]


dev.equo.ide.maven.ListTest.help=[
equo-ide:list
  Description: Lists the p2 dependencies of an Eclipse application.
  Implementation: dev.equo.ide.maven.ListMojo
  Language: java

  Available parameters:

    all
      User property: all
      Lists the id and name of all [categories|features|jars] which meet the
      filter criteria.

    detail
      User property: detail
      Lists properties and requirements for all available versions of the given
      unit id.

    filters
      (no description available)

    format (Default: ASCII)
      User property: format
      Determines output format (can be combined with all other commands).

    installed (Default: false)
      User property: installed
      Lists the jars which were installed.

    installs
      (no description available)

    optional (Default: false)
      User property: optional
      Lists any optional requirements which were not installed.

    p2repos
      (no description available)

    problems (Default: false)
      User property: problems
      Lists any problems with the installed jars.

    raw
      User property: raw
      Lists the raw xml for all available versions of the given unit id.
.
.
[INFO] ------------------------------------------------------------------------
]